server/controller/websocket.py

```diff
@@ -41,19 +41,19 @@ def handle_websocket():
                 continue
 
             decoded_message = json.loads(message)
-            message_type = decoded_message['messageType']
-            if message_type is None:
-                # TODO: blow up
-                pass
+            message_type = decoded_message.get('messageType', None)
+            if message_type not in _ws_routes:
+                print('Unrecognized message_type %s' % message_type)
+                continue
 
             _ws_routes[message_type](decoded_message, websocket,
-                                     _websocket_metadata)
+                                     websocket_metadata)
         except WebSocketError:
             break
 
     # If we have the API key, we can waste a little less time searching for the
     # WebSocket.
-    ws_api_key = _websocket_metadata.get('apiKey', '')
+    ws_api_key = websocket_metadata.get('apiKey', '')
     if (ws_api_key and ws_api_key in _web_ui_ws_connections and
```

Comment:
can be refactored to `if websocket in _web_ui_ws_connections.get(ws_api_key, []):`